common/ferrostar/src/routing_adapters/osrm/models.rs

```diff
@@ -87,14 +87,40 @@ pub struct Annotation {
     /// NOTE: This annotation is not in the official spec, but is a common extension used by Mapbox
     /// and Valhalla.
     #[serde(default, rename = "maxspeed")]
-    #[allow(dead_code)]
-    max_speed: Vec<MaxSpeed>,
+    pub max_speed: Vec<MaxSpeed>,
+}
+
+/// A flattened OSRM flavored annotation object.
+/// This represents the value for each annotation type and is typically surfaced for each
+/// geometry segment.
+#[derive(Serialize, Debug, Clone, PartialEq)]
+pub struct AnnotationValue {
```

Comment:
This is probably a bit too Valhalla-specific. Here is what "actual" OSRM provides: https://project-osrm.org/docs/v5.5.1/api/#annotation-object.

I think these should _all_ be optional fields, and our fields should be a superset of all the known APIs. (I'd also like to have support for arbitrary keys/values too, but that can perhaps come later). For me, that means we need to additionally look at the API docs for OSRM itself, Mapbox, and probably GraphHopper. Though I can't find any docs on what annotations GraphHopper supports. Maybe @hactar can share some details.

Reply:
This no longer exists. Resolving